arangod/Replication/Syncer.cpp

```diff
@@ -150,6 +151,20 @@ arangodb::Result applyCollectionDumpMarkerInternal(
       if (keySlice.isString()) {
         std::pair<arangodb::LocalDocumentId, TRI_voc_rid_t> lookupResult;
         if (coll->getPhysical()->lookupKey(&trx, keySlice.stringRef(), lookupResult).ok()) {
+          // determine if we already have this revision or need to replace the
+          // one we have
+          arangodb::LocalDocumentId docId{
+              arangodb::transaction::helpers::extractRevFromDocument(slice)};
+          if (docId.isSet()) {
+            arangodb::ManagedDocumentResult mdr;
+            if (coll->getPhysical()->readDocument(&trx, docId, mdr)) {
```

Comment:
Mmh, doesn't this slow down every update/replace operation during replication?

Reply:
Yes, during incremental (non-synchronous) replication. But there's not really a different good way to avoid replace-with-self, I think?